### PR TITLE
Fix BSA-122: expose only eligible properties as categories

### PR DESCRIPTION
**Pull request by wazelin**
- [BSA-122](https://oat-sa.atlassian.net/browse/BSA-122) fix: expose only eligible item properties as test-item categories

### CODE REVIEW

File: models/classes/CategoryService.php [hunk 1]
```diff
@@ -132,27 +132,33 @@ public static function sanitizeCategoryName($value)
...
+                    in_array($property->getUri(), self::$excludedPropUris, true)
+                    || !$this->doesExposeCategory($property)
+                ) {
                     return false;
                 }
+
                 $widget = $property->getWidget();
-                return !is_null($widget) && in_array($widget->getUri(), self::$supportedWidgetUris);
+
+                return null !== $widget && in_array($widget->getUri(), self::$supportedWidgetUris, true);
```
Comment (ktsirkou) [inline models/classes/CategoryService.php:153]: out of curiosity why changing from `!is_null` to `null !===` ?

Comment (wazelin) [reply models/classes/CategoryService.php:153]: Just a force of habit. Cannot help myself when I see a function call, which can be replaced with an expression.

**Comment by wazelin** [comment]
> As i was testing, now I cannot render any property as exposable? The button doesn't appear anymore. Can you check please?

You're right. I didn't notice it yesterday, but the "untagged" properties cannot be tagged again anymore now :/

File: models/classes/CategoryService.php (hunk 1 above)
Comment (Alroniks) [reply models/classes/CategoryService.php:150]: The expression also is a faster than function call. :)

**Pull request merged**